### DISCUSSION TITLE
Filter in custom script?

### DISCUSSION THREAD

**Discussion by anders-nc** [netbox-community/netbox#14578] [Help Wanted!] [answered]
Hi,
Found the custom script example https://github.com/netbox-community/netbox-zero-to-hero/blob/main/custom_scripts/NewBranchScript.py 
For creating a new branch, i tried to modify it a bit so that instead of returning all Devicetypes just return a specific manufacturer.
Im not especially good at python i tested lots of variations with filter so im stuck how i should go about i if i just want to see manufacturer = 'Dell' , the main reason is that the list gets awfully long

    access_switch_model = ObjectVar(
        description="Access switch model",
        model=DeviceType
    )
    
    But in principle im after that the object shown for access switches is actually tagged as access switches , ( we have tagged all Devicetypes depending on type)
   
![image](https://github.com/netbox-community/netbox/assets/142779654/429a9f39-d7f3-4027-aeeb-eeda10d469b3)

**Comment by candlerb** [comment, marked as answer]
This is described at https://docs.netbox.dev/en/stable/customization/custom-scripts/#objectvar :

> To limit the selections available within the list, additional query parameters can be passed as the `query_params` dictionary

I believe these are the same query params as you pass to the REST API (i.e. the web page using the REST API to populate the list). 

> we have tagged all Devicetypes depending on type

I would expect you can do (untested):

```
access_switch_model = ObjectVar(
    description="Access switch model",
    model=DeviceType,
    query_params={
        'tag': 'access-switch',  # this is the *slug*, not the *name*, of the tag
    },
)
```

See also https://demo.netbox.dev/api/schema/swagger-ui/#/dcim/dcim_device_types_list for the available query params. (Unfortunately if you give an invalid query param, it's silently ignored as if it didn't exist, so you'll get the full list of device types)

**Reply by anders-nc** [reply]
Thank you :-)

**Comment by anders-nc** [comment]
Thank you :-)
